### ISSUE TITLE
Inline comments striped from stored procedure.... on import

### ISSUE THREAD

**Issue by bkdotcom** [phpmyadmin/phpmyadmin#11717]
``` sql
DROP PROCEDURE IF EXISTS `my_procedure`$$
CREATE DEFINER=`root`@`localhost` PROCEDURE `my_procedure` (IN `_foo` VARCHAR(128))
MODIFIES SQL DATA
BEGIN

    # This procedure is worthless
    SELECT 1 = 1;     # I'm a comment
    # Comments between BEGIN and END blocks should not be stripped 
    #    on import (or via SQL input)

END$$
```

Export exports the comments
**Import** strips the comments

The stored-procedure / "routine" editor preserves the comments

mySQL workbench somehow preserves the comments

Parser issue?  Sorcery?

**Comment by zaverichintan** [comment]
How to reproduce the error?

**Comment by bkdotcom** [comment]
@zaverichintan :  easy..  create a stored procedure that contains inline-comments  (ie the one given in the original bug report).    export the database (note that the export contains the inline comments).
now import the database.    comments are now gone.

[test_db_20160215.sql.zip](https://github.com/phpmyadmin/phpmyadmin/files/131766/test_db_20160215.sql.zip) is a simple test database export... 1 table & 1 stored procedure...
the stored procedure contains all 3 types of inline comments #, --, & /* */
all 3 get stripped when importing

**Comment by moppo24** [comment]
Same issue here. It also happens when creating a procedure via the 'SQL' tab

PhpMyAdmin version: 4.5.2

Does someone know which is the last stable version where this bug is not occurring ?

**Comment by loonighan** [comment]
I've lost a lot of time trying different versions, above and below. Latest version that imports comments is 1.8.3. I upgraded from it because it doesn't support export templates. Wish I hadn't.

**Comment by barrystaes** [comment]
Also had the same problem, after import the comments inside routines are gone.
I am using a new setup with phpMyAdmin 4.6.6 / MariaDB 5.5.54 / PHP 7.1.1 / CentOS 7.
Previous setup without problem: phpMyAdmin 4.3.2 / MySQL 5.6.22 / PHP 5.5.23 / CentOS 6.

**Comment by moppo24** [comment]
someone got a solution for this bug ?

**Comment by ibennetch** [comment]
I'd like if we can fix this before 4.7.0 if there is anyone who wants to fix it before then.

**Comment by loonighan** [comment]
I´m afraid that comments to stored procedures are now the least of my worries. As I was taught at an early age you develop and then put into production. My PhpMyAdmin comes bundled in a xampp distribution. My development suite featured a MySql db server, my production features a MariaDb server. Guess what? UPDATEs & DELETEs are consistent across platforms but alas SELECTs are not. My fault entirely, just another field in ORDER BY clause, in a 150 odd statements. In the meantime a logistics app was a complete disaster.  Moral to the story? If it works, don´t upgrade.

**Comment by nytmo** [comment]
Any chance, this will be fixed after a year? After upgrading to MySQL 5.7, I was forced to use new phpMyAdmin and suddenly lost all the comments in all stored procedures and functions... Please turn off comment stripping during import.

**Comment by Achilles-96** [comment]
The issue is in parser. Try this patch - https://github.com/phpmyadmin/sql-parser/pull/146/files .

**Comment by Achilles-96** [comment]
I think the patch requires some more tweaking. It only works for bash comments now. I will update it.

**Comment by Achilles-96** [comment]
It should be fine. I tested it with all three types of comments.

**Comment by nytmo** [comment]
Great patch. Works like a charm. And now even the export comments problem, that is much more destructive:  #13139

**Comment by nijel** [comment]
Fixed by https://github.com/phpmyadmin/sql-parser/pull/146

**Comment by nijel** [comment]
Fix is part of SQL parser 4.1.3.
